### PR TITLE
Add more logging to PutOperation to debug memory leak

### PR DESCRIPTION
**Pull request by justinlin-linkedin**
Adding more logging information in the PutOperation to debug memory leak.

### CODE REVIEW

File: ambry-router/src/main/java/com/github/ambry/router/PutOperation.java [hunk 1]
```diff
@@ -161,7 +161,7 @@
 
   // Variables to keep track of netty chunks
   private final RestRequest restRequest;
-  private final String touchMessage;
+  private final String loggingContext;
```
Comment (zzmao) [inline ambry-router/src/main/java/com/github/ambry/router/PutOperation.java:164]: Looks like you want to print channel information. How about name it to channelContext ?

File: ambry-router/src/main/java/com/github/ambry/router/PutOperation.java [hunk 2]
```diff
@@ -1771,16 +1781,20 @@ protected PutRequest createPutRequest() {
...
-  private String makeTouchMessage() {
+  private String makeLoggingContext() {
     if (restRequest == null || restRequest.getRestRequestContext() == null) {
       return "";
     }
     String channelStr =
         ((NettyRequest.NettyRequestContext) restRequest.getRestRequestContext()).getChannel().toString();
-    return restRequest.getUri() + " " + channelStr;
+    if (isEncryptionEnabled) {
+      return "True " + channelStr;
```
Comment (zzmao) [inline ambry-router/src/main/java/com/github/ambry/router/PutOperation.java:1794]: What would the return string looks like? 

I feel we can make it more informative and formatted.  

maybe: 
channelStr + " Encrypted: True"

**Review by zzmao — APPROVED** [review]
Code LGTM

File: ambry-router/src/main/java/com/github/ambry/router/PutOperation.java (hunk 1 above)
Comment (justinlin-linkedin) [reply ambry-router/src/main/java/com/github/ambry/router/PutOperation.java:164]: I also include some other information like encryption. And this is per-putoperation context for logging. I think loggingContext is fine.

File: ambry-router/src/main/java/com/github/ambry/router/PutOperation.java (hunk 2 above)
Comment (justinlin-linkedin) [reply ambry-router/src/main/java/com/github/ambry/router/PutOperation.java]: I added Encrypted to the context.

**Review by zzmao — APPROVED** [review]

**Review by cgtz — APPROVED** [review]

**Pull request merged**